src/Init/Data/List/Attach.lean

```diff
@@ -13,29 +13,45 @@ set_option linter.indexVariables true -- Enforce naming conventions for index va
 
 namespace List
 
-/-- `O(n)`. Partial map. If `f : Π a, P a → β` is a partial function defined on
-  `a : α` satisfying `P`, then `pmap f l h` is essentially the same as `map f l`
-  but is defined only when all members of `l` satisfy `P`, using the proof
-  to apply `f`. -/
+/--
+Maps a function from the elements of `α` that satisfy `P` to `β` over the list `l`, given a proof
+that every element of `l` in fact satisfies `P`.
+
+`O(|l|)`. The function `f` is a partial function, defined only for those elements of `α` that
+satisfy `P`. `List.pmap`, named for “partial map,” is the equivalent of `List.map` for such partial
+functions.
```

Comment:
Duplicative now?

Reply:
The first sentence yes, the second no (I think the name deserves a bit of explanation). I'll delete the first.